src/utils/SocketClient.hxx

```diff
@@ -355,10 +426,18 @@ private:
     /** callback to call on connection status */
     std::function<void(Status)> statusCallback_ = nullptr;
     
-        /** number of seconds between retries */
+    /** number of seconds between retries */
     uint8_t retrySeconds_;
+
+    /** time in seconds that the connect is supposed to
+     *  timeout and look for a possible shutdown
+     */
     uint8_t timeoutSeconds_;
 
+    /** disallow local connections to one's self */
+    bool disallowLocal_;
```

Comment:
`uint8_t   disallowLocal_ : 1` ?
  one bit would be enough for this and you have some unused alignment bytes here anyway.